src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java

```diff
@@ -182,112 +173,11 @@ public void close() throws IOException {
-     * bytes don't exist this will throw.
-     */
-    private static int getHeaderSize(byte[] bytes) {
-        return bytes[5] * 256 + bytes[6];
-    }
-
-    /**
-     * Check if an IronCore header is present in some bytes, indicating that it is ciphertext.
-     *
-     * @param bytes bytes to be checked
-     */
-    public static boolean isCiphertext(byte[] bytes) {
-        // Header size is variable for CMK encrypted docs depending on whether
-        // the header is present. Expect at least one byte following the header
-        // that would have been encrypted.
-        return bytes.length > DOCUMENT_HEADER_META_LENGTH
-                && bytes[0] == CURRENT_DOCUMENT_HEADER_VERSION && containsIroncoreMagic(bytes)
-                && getHeaderSize(bytes) >= 0;
-    }
-
-    /**
-     * Parses the header off the encrypted document and returns a ByteBuffer wrapping the document
-     * bytes. Once the header contains metadata we care about, this will return a class containing
-     * the document bytes and the header.
-     */
-    private static CompletableFuture<ByteBuffer> parseDocumentParts(byte[] document) {
-        return CompletableFutures.tryCatchNonFatal(() -> {
-            if (!isCiphertext(document)) {
-                throw new IllegalArgumentException(
-                        "Provided bytes were not an Ironcore encrypted document.");
-            }
-            int totalHeaderSize = getHeaderSize(document) + DOCUMENT_HEADER_META_LENGTH;
-            int newLength = document.length - totalHeaderSize;
-            return ByteBuffer.wrap(document, totalHeaderSize, newLength);
-        });
-    }
-
-    /**
-     * Given the provided encrypted document (which has an IV prepended to it) and an AES key,
-     * decrypt and return the decrypted bytes.
-     */
-    private CompletableFuture<byte[]> decryptBytes(ByteBuffer encryptedDocument,
-            byte[] documentKey) {
-        byte[] iv = new byte[IV_BYTE_LENGTH];
-
-        // Pull out the IV from the front of the encrypted data
-        encryptedDocument.get(iv);
-        byte[] encryptedBytes = new byte[encryptedDocument.remaining()];
-        encryptedDocument.get(encryptedBytes);
-
-        return CompletableFutures.tryCatchNonFatal(() -> {
-            final Cipher cipher = Cipher.getInstance(AES_ALGO);
-            cipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(documentKey, "AES"),
-                    new GCMParameterSpec(GCM_TAG_BIT_LENGTH, iv));
-            return cipher.doFinal(encryptedBytes);
-        });
-    }
-
     /**
      * Encrypt the provided map of fields using the provided encryption key (DEK) and return the
      * resulting encrypted fields in a map from String to encrypted bytes.
      */
-    private Map<String, byte[]> encryptFields(Map<String, byte[]> document, byte[] dek) {
+    private Map<String, byte[]> encryptFields(Map<String, byte[]> document,DocumentMetadata metadata, byte[] dek) {
```

Comment:
I'll probably figure this out as I read, but why did `DocumentMetadata` get added here?